src/subcommands/common.rs

```diff
@@ -15,6 +19,23 @@ pub struct ProviderOpts {
     pub provider: u8,
 }
 
+impl ProviderOpts {
+    /// Get the ProviderID selected by the user or the service default if
+    /// no provider was selected.
+    ///
+    /// The Core Provider cannot be used and will be overriden by the
+    /// service default.
+    pub fn provider(&self) -> Result<ProviderID, ParsecToolError> {
+        if self.provider != 0 {
+            Ok(ProviderID::try_from(self.provider)?)
```

Comment:
It's correct for now, but should we convert the `u8` to a `ProviderID`? We had this idea that in the future, providers could have *any* ID...

Reply:
Oh we might have to because the `process_operation` method takes a `ProviderID`

Reply:
I'm aware, but it felt more reasonable for `ProviderOpts` to return the final type instead of always parsing it in the subcomand. We can just change this when we do the rest of the work - it's a 1 line change now, instead of 1 per file :P